ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -422,27 +414,25 @@ func (r *RayJobReconciler) releaseComputeResources(ctx context.Context, rayJobIn
 	// Since the name of the Kubernetes Job is the same as the RayJob, we need to delete the Kubernetes Job
 	// and its Pods when suspending. A new submitter Kubernetes Job must be created to resubmit the
 	// Ray job if the RayJob is resumed.
-	if isSuspend {
-		jobName := rayJobInstance.Name
-		job := &batchv1.Job{}
-		if err := r.Client.Get(ctx, client.ObjectKey{Namespace: namespace, Name: jobName}, job); err != nil {
-			if errors.IsNotFound(err) {
-				isJobNotFound = true
-				r.Log.Info("The submitter Kubernetes Job has been already deleted", "RayJob", rayJobInstance.Name, "Kubernetes Job", job.Name)
-			} else {
-				r.Log.Error(err, "Failed to get Kubernetes Job")
-				return false, err
-			}
+	jobName := rayJobInstance.Name
+	job := &batchv1.Job{}
+	if err := r.Client.Get(ctx, client.ObjectKey{Namespace: namespace, Name: jobName}, job); err != nil {
+		if errors.IsNotFound(err) {
+			isJobNotFound = true
+			r.Log.Info("The submitter Kubernetes Job has been already deleted", "RayJob", rayJobInstance.Name, "Kubernetes Job", job.Name)
 		} else {
-			if !job.DeletionTimestamp.IsZero() {
-				r.Log.Info("The Job deletion is ongoing.", "RayJob", rayJobInstance.Name, "Submitter K8s Job", job.Name)
-			} else {
-				if err := r.Client.Delete(ctx, job, client.PropagationPolicy(metav1.DeletePropagationBackground)); err != nil {
-					return false, err
-				}
-				r.Log.Info("The associated submitter Job is deleted", "RayJob", rayJobInstance.Name, "Submitter K8s Job", job.Name)
-				r.Recorder.Eventf(rayJobInstance, corev1.EventTypeNormal, "Deleted", "Deleted submitter K8s Job %s", job.Name)
+			r.Log.Error(err, "Failed to get Kubernetes Job")
+			return false, err
+		}
+	} else {
+		if !job.DeletionTimestamp.IsZero() {
+			r.Log.Info("The Job deletion is ongoing.", "RayJob", rayJobInstance.Name, "Submitter K8s Job", job.Name)
+		} else {
+			if err := r.Client.Delete(ctx, job, client.PropagationPolicy(metav1.DeletePropagationBackground)); err != nil {
```

Comment:
For future consideration: since we are controlling the lifecycle of the submitter job, it might be worthwhile to allow users to tune the TTL of only the submitter job. This way you can keep the logs from the job around without having to keep the cluster around.

Reply:
(a common problem with RayJobs that are immediately deleted is that it can be difficult to troubleshoot long running jobs because everything is cleaned up immedaitely)

Reply:
> For future consideration: since we are controlling the lifecycle of the submitter job, it might be worthwhile to allow users to tune the TTL of only the submitter job. This way you can keep the logs from the job around without having to keep the cluster around.

This makes sense to me. Are you interested in opening a PR for this after this PR got merged? Thanks!

Reply:
Sure thing!

Reply:
Thanks! Would you mind opening an issue for it, so that I can assign it to you? (If I open the issue, I can't assign it to you unless you comment on it.)

Reply:
https://github.com/ray-project/kuberay/issues/1832